### PR TITLE
Feature/fix alpha test

### PR DESCRIPTION
**Pull request by Mee-gu**
fix the alpha test

### CODE REVIEW

File: cocos/2d/CCNode.cpp [hunk 1]
```diff
@@ -2233,4 +2175,16 @@ int Node::getAttachedNodeCount()
     return __attachedNodeCount;
 }
 
+void Node::setProgramState(backend::ProgramState* programState)
+{
+    CC_SAFE_RELEASE(_programState);
+    _programState = programState;
+    CC_SAFE_RETAIN(programState);
```
Comment (minggo) [inline cocos/2d/CCNode.cpp]: should retain programState first, or if programState equals _ programState, then may have problem

Comment (minggo) [reply cocos/2d/CCNode.cpp]: or should check if programState equals _ programState

File: cocos/2d/CCSprite.h [hunk 2]
```diff
@@ -420,12 +420,12 @@ class CC_DLL Sprite : public Node, public TextureProtocol
...
     * Set ProgramState
     */
-    void setProgramState(backend::ProgramState *programState);
+    virtual void setProgramState(backend::ProgramState *programState) override;
 
     /**
     * Get current ProgramState
     */
-    inline backend::ProgramState *getProgramState() const { return _programState; }
+    virtual inline backend::ProgramState *getProgramState() const override { return _programState; }
```
Comment (minggo) [inline cocos/2d/CCSprite.h]: Can virtual function be inline?

**Comment by Mee-gu** [comment]
update review

File: cocos/scripting/lua-bindings/auto/lua_cocos2dx_studio_auto.cpp [hunk 3]
```diff
@@ -9530,6 +9530,53 @@ int lua_register_cocos2dx_studio_ArmatureAnimation(lua_State* tolua_S)
     return 1;
 }
 
+int lua_cocos2dx_studio_ArmatureDataManager_getAnimationDatas(lua_State* tolua_S)
```
Comment (minggo) [inline cocos/scripting/lua-bindings/auto/lua_cocos2dx_studio_auto.cpp:9533]: Why modify the binding codes?

Comment (Mee-gu) [reply cocos/scripting/lua-bindings/auto/lua_cocos2dx_studio_auto.cpp:9533]: It was auto-generated by genbindings.py

**Review by minggo — APPROVED** [review]

**Pull request merged**